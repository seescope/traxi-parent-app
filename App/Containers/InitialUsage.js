// @flow

import React from 'react';
import { Text, StyleSheet, View, ScrollView } from 'react-native';
import _ from 'lodash';
import shortid from 'shortid';
import { connect } from 'react-redux';
import { Actions } from 'react-native-router-flux';
import getInitialUsage from '../AsyncActions/GetInitialUsage';
import Spacing from '../Components/Spacing';
import Background from '../Components/Background';
import HeaderText from '../Components/HeaderText';
import BodyText from '../Components/BodyText';
import { firstName } from '../Utils';
import AppRow from '../Components/InitialUsage/AppRow';
import LoadingIndicator from '../Components/LoadingIndicator';
import { LIGHT_GREY, GREY, WHITE, SHADOW_COLOR } from '../Constants/Colours';
import { fetchingApps } from '../Reducers/Setup/setupActions';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    paddingTop: 40,
  },
  cardContainer: {
    flex: 1,
    alignSelf: 'stretch',
    borderRadius: 8,
    backgroundColor: WHITE,
    margin: 10,
    padding: 8,
    marginBottom: 28,
    shadowColor: SHADOW_COLOR,
    shadowOpacity: 0.16,
    shadowRadius: 3,
    shadowOffset: {
      height: 3,
      width: 3,
    },
  },
  cardBody: {
    margin: 10,
  },
  cardHeaderStyle: {
    padding: 8,
    color: GREY,
    fontSize: 19,
  },
  headerContainer: {
    height: 40,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  headerUnderlineStyle: {
    marginHorizontal: 9,
    height: 1,
    width: 60,
    backgroundColor: LIGHT_GREY,
  },
});

console.disableYellowBox = true;

const getHeaderLabel = (count, usageIsNotComplete) => {
  if (!usageIsNotComplete) return 'Completed';
  else if (count === 1) return `${count} app found`;
  else if (count > 1) return `${count} apps found`;
  return 'Searching for apps...';
};

const checkIfUsageIsComplete = apps => {
  if (apps.length < 3) return false;
  return apps.every(app => app.progress === 100);
};

const sleep = milliseconds =>
  new Promise(resolve => setTimeout(resolve, milliseconds));

type App = {
  name: string,
  logo: string,
  progress: number,
};

class InitialUsageComponent extends React.Component {
  state = {
    usageIsNotComplete: true,
    progressHistory: {},
  };

  componentDidMount() {
    console.log('MOUUUUUNT');
    this.props.reset();
    this.updateUsage();

    //TO remove
    // setTimeout(
    //   () => {
    //     this.props.upd();
    //     setTimeout(
    //       () => {
    //         !this.props.upd();
    //       },
    //       2000,
    //     );
    //   },
    //   100,
    // );
  }

  componentWillReceiveProps() {
    const apps = this.props.apps;

    if (apps != null) {
      const progressHistory = apps.reduce(
        (previous, app) => ({
          ...previous,
          [app.name]: app.progress,
        }),
        {},
      );

      this.setState({ progressHistory });
    }
  }

  getPreviousProgress(name) {
    const progressHistory = this.state.progressHistory[name];
    if (progressHistory) return progressHistory;
    return 0;
  }

  updateUsage() {
    if (this.props.isFetchingApps) {
      sleep(500).then(() => this.updateUsage());
      return;
    }

    const usageIsComplete = checkIfUsageIsComplete(this.props.apps);

    if (!usageIsComplete) {
      this.props.getApps();
      sleep(2000).then(() => this.updateUsage());
      return;
    }

    sleep(1500).then(() => {
      Actions.setupCompletion();
      this.setState({
        usageIsNotComplete: false,
      });
    });
  }

  props: {
    kidName: string,
    apps: App[],
    isFetchingApps: boolean,
    getApps: () => void,
  };

  renderAppRows = apps => {
    if (!apps.length) {
      return <AppRow name={'Application name'} progress={0} logo={''} />;
    }

    return apps.map(app => {
      const { name, progress, logo } = app;

      const previousProgress = this.getPreviousProgress(name);

      return (
        <AppRow
          key={shortid.generate()}
          name={name}
          previousProgress={previousProgress}
          progress={progress}
          logo={logo}
        />
      );
    });
  };

  render() {
    const { kidName, apps } = this.props;

    const { usageIsNotComplete } = this.state;

    return (
      <Background>
        <View style={styles.container}>
          <ScrollView>
            <HeaderText>
              Let's see some usage!
            </HeaderText>

            <Spacing height={32} />

            <BodyText>
              {kidName}
              , Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
              eiusmod
              {'\n\n'}
              Ut enim ad minim veniam, quis nostrud exercitation ullamco
              {'\n\n'}
              Duis aute irure dolor in reprehenderit in voluptate velit esse
              cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
              cupidatat non
            </BodyText>

            <Spacing height={16} />

            <View style={styles.cardContainer}>
              <View style={styles.headerContainer}>
                <Text style={styles.cardHeaderStyle}>
                  {getHeaderLabel(apps.length, usageIsNotComplete)}
                </Text>
                {this.state.usageIsNotComplete && <LoadingIndicator />}
              </View>
              <View style={styles.headerUnderlineStyle} />
              <View style={styles.cardBody}>
                {this.renderAppRows(apps)}
              </View>
            </View>
          </ScrollView>
        </View>
      </Background>
    );
  }
}

const getProgressValue = timeUsed => {
  if (timeUsed >= 60) return 100;
  const percentage = timeUsed / 60 * 100;
  return Math.round(percentage);
};

const getAppsInformations = apps =>
  apps.map(app => ({
    name: app.Name,
    logo: app.Logo,
    progress: getProgressValue(app.TimeUsed),
  }));

const mapStateToProps = state => {
  const { apps, kidUUID, isFetchingApps } = state.setupState;
  const { name } = state.kidsState[kidUUID] || {};

  // console.log(JSON.stringify(apps));
  let appsInformation;
  if (!apps) {
    appsInformation = [];
  } else {
    const filterShortTimeUsedApps = apps.filter(app => app.TimeUsed > 5);
    const groupedApps = _.reduce(
      filterShortTimeUsedApps,
      (previous, app) => {
        const index = _.findIndex(previous, ['Name', app.Name]);
        if (index !== -1) {
          const previousToModify = [...previous];
          previousToModify[index].TimeUsed =
            previousToModify[index].TimeUsed + app.TimeUsed;
          return previousToModify;
        }
        return [...previous, app];
      },
      [],
    );
    appsInformation = getAppsInformations(groupedApps);
  }

  return {
    apps: appsInformation,
    kidName: name ? firstName(name) : '',
    isFetchingApps,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    //TO REMOVE
    reset: () => dispatch(fetchingApps(false)),
    getApps: () => dispatch(getInitialUsage()),
  };
};

const InitialUsage = connect(mapStateToProps, mapDispatchToProps)(
  InitialUsageComponent,
);

export default InitialUsage;
