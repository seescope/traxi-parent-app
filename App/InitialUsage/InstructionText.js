// @flow
import React from 'react';
import { View } from 'react-native';
import Spacing from '../Components/Spacing';
import BodyText from '../Components/BodyText';

type Props = {
  kidName: string
};

export default ({ kidName }: Props) => (
  <View>
    <BodyText>
      Now that
      {' '}
      {kidName}
      &apos;s device is being monitored by traxi, you can see what apps they&apos;re using on their device.
    </BodyText>

    <Spacing />

    <BodyText>
      After
      {' '}
      {kidName}
      {' '}
      has used an app for more than a 60 seconds, you'll be able to see it with Traxi.
    </BodyText>

    <Spacing />

    <BodyText>
      To show you how this works, open an app on
      {' '}
      {kidName}
      &apos;s device and use it until the bar turns green.
    </BodyText>

    <Spacing height={32} />
  </View>
);
