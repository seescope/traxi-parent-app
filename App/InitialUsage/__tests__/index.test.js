import React from 'react';
import InitialUsage from '../';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

it('renders correctly when there are no apps', () => {
  const mockStore = configureStore();
  const testStore = mockStore({
    kidsState: {
      'abc-123': {
        name: 'Jim Bob',
      },
    },
    setupState: {
      kidUUID: 'abc123',
      apps: [],
    },
  });

  const tree = renderer.create(
    <Provider store={testStore}><InitialUsage /></Provider>
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders correctly when there is one app', () => {
  const mockStore = configureStore();
  const testStore = mockStore({
    kidsState: {
      'abc-123': {
        name: 'Jim Bob',
      },
    },
    setupState: {
      kidUUID: 'abc123',
      apps: [
        {
          name: 'Trello',
          logo: 'something',
          progress: 10,
        },
      ],
    },
  });

  const tree = renderer.create(
    <Provider store={testStore}><InitialUsage /></Provider>
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders correctly when there are two apps', () => {
  const mockStore = configureStore();
  const testStore = mockStore({
    kidsState: {
      'abc-123': {
        name: 'Jim Bob',
      },
    },
    setupState: {
      kidUUID: 'abc123',
      apps: [
        {
          name: 'Trello',
          logo: 'something',
          progress: 10,
        },
        {
          name: 'Not Trello',
          logo: 'something',
          progress: 10,
        },
      ],
    },
  });

  const tree = renderer.create(
    <Provider store={testStore}><InitialUsage /></Provider>
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
