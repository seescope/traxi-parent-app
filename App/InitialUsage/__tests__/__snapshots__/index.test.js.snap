// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no apps 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F0F2F7",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "paddingTop": 32,
      }
    }
  >
    <RCTScrollView>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#545454",
              "fontFamily": "Raleway-ExtraBold",
              "fontSize": 26,
              "textAlign": "center",
            }
          }
        >
          Let's see some apps!
        </Text>
        <View
          style={
            Object {
              "height": 32,
            }
          }
        />
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#545454",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 16,
                  "paddingHorizontal": 22,
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            Now that
             
            
            's device is being monitored by traxi, you can see what apps they're using on their device.
          </Text>
          <View
            style={
              Object {
                "height": 24,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#545454",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 16,
                  "paddingHorizontal": 22,
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            After
             
            
             
            has used an app for more than a 60 seconds, you'll be able to see it with Traxi.
          </Text>
          <View
            style={
              Object {
                "height": 24,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#545454",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 16,
                  "paddingHorizontal": 22,
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            To show you how this works, open an app on
             
            
            's device and use it until the bar turns green.
          </Text>
          <View
            style={
              Object {
                "height": 32,
              }
            }
          />
        </View>
        <View
          animation="bounceInUp"
          duration={1000}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 4,
              "elevation": 6,
              "margin": 8,
              "marginBottom": 24,
              "padding": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 0.16,
              "shadowRadius": 3,
              "width": 734,
            }
          }
          useNativeDriver={true}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#545454",
                  "fontFamily": "Raleway-ExtraBold",
                  "fontSize": 17,
                  "padding": 8,
                  "paddingBottom": 0,
                }
              }
            >
              Searching for apps...
            </Text>
            <View>
              <ActivityIndicator
                animating={true}
                color="#448AFF"
                hidesWhenStopped={true}
                size="large"
                style={null}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A6A8AB",
                "height": 0.5,
                "marginHorizontal": 9,
                "marginTop": -8,
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "margin": 10,
              }
            }
          >
            <View
              animation="fadeIn"
              duration={1000}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 8,
                }
              }
              useNativeDriver={true}
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": 40,
                    "marginRight": 8,
                    "width": 40,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#A6A8AB",
                      "borderRadius": 8,
                      "flex": 1,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#545454",
                        "fontFamily": "Raleway-ExtraBold",
                        "marginBottom": 4,
                        "marginRight": 8,
                      }
                    }
                  >
                    Loading...
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#545454",
                        "fontFamily": "Raleway-Regular",
                        "fontWeight": "300",
                      }
                    }
                  >
                    0
                    %
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#A6A8AB",
                      "borderRadius": 4,
                      "flex": 1,
                      "height": 20,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#448AFF",
                        "borderRadius": 4,
                        "flex": 1,
                        "width": "0%",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          animation="bounceInUp"
          delay={1000}
          duration={1000}
          useNativeDriver={true}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#545454",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 16,
                  "paddingHorizontal": 22,
                  "textAlign": "left",
                },
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Use an app on 's device
            </Text>
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 16,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#448AFF",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 14,
                }
              }
            >
              [missing "en.general.needHelp" translation]
               
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`renders correctly when there are two apps 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F0F2F7",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "paddingTop": 32,
      }
    }
  >
    <RCTScrollView>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#545454",
              "fontFamily": "Raleway-ExtraBold",
              "fontSize": 26,
              "textAlign": "center",
            }
          }
        >
          Let's see some apps!
        </Text>
        <View
          style={
            Object {
              "height": 32,
            }
          }
        />
        <View
          animation="bounceInUp"
          duration={1000}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 4,
              "elevation": 6,
              "margin": 8,
              "marginBottom": 24,
              "padding": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 0.16,
              "shadowRadius": 3,
              "width": 734,
            }
          }
          useNativeDriver={true}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#545454",
                  "fontFamily": "Raleway-ExtraBold",
                  "fontSize": 17,
                  "padding": 8,
                  "paddingBottom": 0,
                }
              }
            >
              Found 2 apps
            </Text>
            <View>
              <ActivityIndicator
                animating={true}
                color="#448AFF"
                hidesWhenStopped={true}
                size="large"
                style={null}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A6A8AB",
                "height": 0.5,
                "marginHorizontal": 9,
                "marginTop": -8,
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "margin": 10,
              }
            }
          >
            <View
              animation="fadeIn"
              duration={1000}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 8,
                }
              }
              useNativeDriver={true}
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": 40,
                    "marginRight": 8,
                    "width": 40,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "something",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 8,
                      "flex": 1,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#545454",
                        "fontFamily": "Raleway-ExtraBold",
                        "marginBottom": 4,
                        "marginRight": 8,
                      }
                    }
                  >
                    Trello
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#545454",
                        "fontFamily": "Raleway-Regular",
                        "fontWeight": "300",
                      }
                    }
                  >
                    10
                    %
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#A6A8AB",
                      "borderRadius": 4,
                      "flex": 1,
                      "height": 20,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#448AFF",
                        "borderRadius": 4,
                        "flex": 1,
                        "width": "0%",
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              animation="fadeIn"
              duration={1000}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 8,
                }
              }
              useNativeDriver={true}
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": 40,
                    "marginRight": 8,
                    "width": 40,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "something",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 8,
                      "flex": 1,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#545454",
                        "fontFamily": "Raleway-ExtraBold",
                        "marginBottom": 4,
                        "marginRight": 8,
                      }
                    }
                  >
                    Not Trello
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#545454",
                        "fontFamily": "Raleway-Regular",
                        "fontWeight": "300",
                      }
                    }
                  >
                    10
                    %
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#A6A8AB",
                      "borderRadius": 4,
                      "flex": 1,
                      "height": 20,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#448AFF",
                        "borderRadius": 4,
                        "flex": 1,
                        "width": "0%",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          animation="bounceInUp"
          delay={1000}
          duration={1000}
          useNativeDriver={true}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#545454",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 16,
                  "paddingHorizontal": 22,
                  "textAlign": "left",
                },
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Keep using Trello until the bar turns green
            </Text>
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 16,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#448AFF",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 14,
                }
              }
            >
              [missing "en.general.needHelp" translation]
               
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`renders correctly when there is one app 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F0F2F7",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "paddingTop": 32,
      }
    }
  >
    <RCTScrollView>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#545454",
              "fontFamily": "Raleway-ExtraBold",
              "fontSize": 26,
              "textAlign": "center",
            }
          }
        >
          Let's see some apps!
        </Text>
        <View
          style={
            Object {
              "height": 32,
            }
          }
        />
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#545454",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 16,
                  "paddingHorizontal": 22,
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            Now that
             
            
            's device is being monitored by traxi, you can see what apps they're using on their device.
          </Text>
          <View
            style={
              Object {
                "height": 24,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#545454",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 16,
                  "paddingHorizontal": 22,
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            After
             
            
             
            has used an app for more than a 60 seconds, you'll be able to see it with Traxi.
          </Text>
          <View
            style={
              Object {
                "height": 24,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#545454",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 16,
                  "paddingHorizontal": 22,
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            To show you how this works, open an app on
             
            
            's device and use it until the bar turns green.
          </Text>
          <View
            style={
              Object {
                "height": 32,
              }
            }
          />
        </View>
        <View
          animation="bounceInUp"
          duration={1000}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 4,
              "elevation": 6,
              "margin": 8,
              "marginBottom": 24,
              "padding": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 0.16,
              "shadowRadius": 3,
              "width": 734,
            }
          }
          useNativeDriver={true}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#545454",
                  "fontFamily": "Raleway-ExtraBold",
                  "fontSize": 17,
                  "padding": 8,
                  "paddingBottom": 0,
                }
              }
            >
              Found 1 app
            </Text>
            <View>
              <ActivityIndicator
                animating={true}
                color="#448AFF"
                hidesWhenStopped={true}
                size="large"
                style={null}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A6A8AB",
                "height": 0.5,
                "marginHorizontal": 9,
                "marginTop": -8,
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "margin": 10,
              }
            }
          >
            <View
              animation="fadeIn"
              duration={1000}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 8,
                }
              }
              useNativeDriver={true}
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": 40,
                    "marginRight": 8,
                    "width": 40,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "something",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 8,
                      "flex": 1,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#545454",
                        "fontFamily": "Raleway-ExtraBold",
                        "marginBottom": 4,
                        "marginRight": 8,
                      }
                    }
                  >
                    Trello
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#545454",
                        "fontFamily": "Raleway-Regular",
                        "fontWeight": "300",
                      }
                    }
                  >
                    10
                    %
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#A6A8AB",
                      "borderRadius": 4,
                      "flex": 1,
                      "height": 20,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#448AFF",
                        "borderRadius": 4,
                        "flex": 1,
                        "width": "0%",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          animation="bounceInUp"
          delay={1000}
          duration={1000}
          useNativeDriver={true}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#545454",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 16,
                  "paddingHorizontal": 22,
                  "textAlign": "left",
                },
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Keep using Trello until the bar turns green
            </Text>
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 16,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#448AFF",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 14,
                }
              }
            >
              [missing "en.general.needHelp" translation]
               
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
